### DISCUSSION TITLE
At what point (when opening a new terminal) is ~/.zprofile read?

### DISCUSSION THREAD

**Discussion by Disco-Chef** [ohmyzsh/ohmyzsh#10206] [Help] [answered]
What triggers it from the ~/.zshrc?

Just trying to figure out why the pyenv plugin is telling me that the path is not set right for pyenv, when i've put the lines it wants:

```
export PYENV_ROOT="$HOME/.pyenv"
export PATH="$PYENV_ROOT/bin:$PATH"
eval "$(pyenv init --path)"

```
in my ~/.zprofile, and yet it is still warning that pyenv path is not configured well. pyenv works fine though (is is indeed in the path, just not the moment that pyenv plugin check, but later i assume)

**Comment by mcornella** [comment, marked as answer]
`.zprofile` is ran in a login shell before `.zshrc` (see [startup files docs](https://zsh.sourceforge.io/Intro/intro_3.html)). You should also add those lines to `.profile` though, and make sure to restart your user session so the changes take effect.

**Reply by Disco-Chef** [reply]
Thank you for answering.
Mind if I bug you just a little bit more? 😅 

If those lines are in `.zprofile`, why the need to also add them to `.profile` ?  Previously, I had those lines only in `.zprofile` and I was not getting the warning (on ubuntu).
I don't mind having those lines also in `.profile`, I'm just trying to understand what was/is going on i.e. on the order of everything at startup. If `.zprofile` is read before `.zshrc`, then I thought I shouldn't be getting that warning.

On the other hand, with the same setup but in a WSL2 ubuntu shell, I do not get the warning (and there I also have those lines only in `.zprofile` and not `.profile` 🤔

**Reply by mcornella** [reply]
In some platforms or terminal emulators zsh is not started as a login shell, which is a precondition for `.zprofile` to be read. If you add this to `.profile`, and restart your user session, `.profile` is read and therefore the environment will be updated for all of your user session, including whether or not you're running in an interactive shell or in a script.

**Reply by Disco-Chef** [reply]
Coolio. So i'm guessing that on ubuntu zsh is not started as a login shell, whereas on WSL it is. Thanks!
